archaius2-core/src/main/java/com/netflix/archaius/ConfigProxyFactory.java

```diff
@@ -189,6 +202,8 @@ interface PropertyValueGetter<T> {
 
     @SuppressWarnings({"unchecked", "rawtypes"})
     <T> T newProxy(final Class<T> type, final String initialPrefix, boolean immutable) {
+        warnWhenTooMany(PROXIES_COUNT, type, 50, String.format("Proxy(%s)", type));
```

Comment:
50 seems a little high for this in general, is this to accommodate the possibility of someone having the same proxy class but with different prefixes? If so, would it make sense to combine that into the key?

Reply:
Yeah, that's the thought.